### PR TITLE
Finished main mutations and other quite important edits...

### PR DESCRIPTION
**Pull request by Braden50**
Finished group mutations and added achievement mutation. Added ability to save images. Updated userMetrics attributes. Added achievement monitoring.

Finished large mutations (still have smaller and simple ones to make to ease the work on the frontend) so the frontend should be able to create, edit, and delete everything in the database (just remember the error conditions I mentioned in my last PR). I added a userMetrics threshold to achievements to check if a user has earned an achievement or not. Additionally, I added a script that runs whenever a user is created or updated to add new achievements if the user has met the threshold and doesn't already have the achievement (could be replaced but as of now we don't have to worry about assigning achievements as long as the thresholds are set correctly and achievements correlate with userMetrics).

Finally, I added the ability to upload images for both users ( profile_pic ) and artworks ( pictures.append(image_to_add) ). The frontend has to handle the encoding and decoding, but the saving and accessing works fine on the backend.  For testing, I added a sample profile pic as a png to the frontend/src/assets. Currently, for the testing and local mock databases, all users have the same sample profile pic and all artworks have the same sample artwork pic. I can add additional pics so we have distinct images; but as of now, the frontend can practice fetching, displaying, and updating the images (except currently you can't remove images from an artwork because I expect the frontend to hand me an index for that and we haven't discussed that yet).

To summarize, the backend is ready for whatever demonstration we want to choreograph for Monday. We would have to add more data to the database.py file for that though unless you guys wanna use my testing setup. This PR solves issues #12 and #20.

### CODE REVIEW

**Review by Waidhoferj — APPROVED** [review]
Looks great! Can't wait to get this hooked up on the frontend!
